apps/store/themes/store/js/list_assets.js

```
@@ -490,114 +490,116 @@ function resolveDomain() {
-                    var taxaSub = [];
-                    $.ajax({
-                        url: caramel.context + '/apis/taxonomies?terms=' + path + "/children" + resolveDomain(),
-                        type: 'GET',
-                        async: false,
-                        headers: {
-                            Accept: "application/json"
-                        },
-                        success: function (data) {
-                            var children;
-                            try {
-                                children = Array.isArray(data[0].children);
-                            } catch (e) {
 
-                            }
+                },
+                error: function () {
+                }
+            });
+
+            // bellow code block will generate sub categories for only
+            if (assetAvailability || store.listAssetsCount > 0) {
+                var nodes = getURL().split("/");
+
+                var path = nodes[0];
+
+                for (var i = 1; i < nodes.length; i++) {
+                    path += "/" + nodes[i];
+                    var elementRef = document.getElementById(path);
+                    // check wheather its leafnode or not
+                    if ($(elementRef).attr("children") == "true") {
+                        var taxaSub = [];
+                        $.ajax({
+                            url: caramel.context + '/apis/taxonomies?terms=' + path + "/children" + resolveDomain(),
+                            type: 'GET',
+                            async: false,
+                            headers: {
+                                Accept: "application/json"
+                            },
+                            success: function (data) {
+                                var children;
+                                try {
+                                    children = Array.isArray(data[0].children);
+                                } catch (e) {
 
```

empty catch block. if don't need to handle, log the error